packages/labs/react/src/create-component.ts

```diff
@@ -16,84 +30,77 @@ const reservedReactProperties = new Set([
 
 const listenedEvents: WeakMap<
   Element,
-  Map<string, EventListenerObject>
+  Map<string, EventListener>
 > = new WeakMap();
 
 /**
- * Adds an event listener for the specified event to the given node. In the
- * React setup, there should only ever be one event listener. Thus, for
- * efficiency only one listener is added and the handler for that listener is
- * updated to point to the given listener function.
+ * Adds an event listener for the specified event to a given node.
+ *
+ * React expects one event listener per event. To mirror this behavior,
+ * the most recent event listeners are cached in listenedEvents.
+ * When an event listener updates, the previously cached event listener
+ * is removed before adding the updated event listener.
  */
 const addOrUpdateEventListener = (
   node: Element,
   event: string,
-  listener: (event?: Event) => void
+  listener: EventListener
 ) => {
   let events = listenedEvents.get(node);
   if (events === undefined) {
     listenedEvents.set(node, (events = new Map()));
   }
-  let handler = events.get(event);
-  if (listener !== undefined) {
-    // If necessary, add listener and track handler
-    if (handler === undefined) {
-      events.set(event, (handler = {handleEvent: listener}));
-      node.addEventListener(event, handler);
-      // Otherwise just update the listener with new value
-    } else {
-      handler.handleEvent = listener;
-    }
-    // Remove listener if one exists and value is undefined
-  } else if (handler !== undefined) {
+
+  // Remove listener if one exists
+  const handler = events.get(event);
+  if (handler !== undefined) {
```

Comment:
The point of the current code was to avoid thrashing the DOM with `addEventListener`/`removeEventListener` calls when fresh closures are passed on repeat renders.

Imagine this JSX:

```tsx
<SomeWebComponent onFoo={(e) => console.log(e)}></SomeWebCompoent>
```

Every render that will pass a new function to `SomeWebComponent.onFoo`. With the current code `addEventListener` is only called once.

Reply:
Nice! I was trying to mimic that behavior but i guesss i biffed it.

The aim was to leverage the diff check in `create-component.ts` line 81. My logic was, if there was an update to the event listener, we'd get pass that diff check and remove / add the new event listener.

Reply:
I think accounting for repeat closures is not 'react-like' but i'm open about this. My rationale is: If a user relies on currying data in a closure, but we don't update those closures, their closure is stale.

Also in a react review I might say:

`<SomeWebComponent onFoo={(e) => console.log(e)}></SomeWebCompoent>`

should be

```
const logger = createCallback((e) => console.log(e)), []);

return (
  <SomeWebComponent onFoo={(e) => console.log(e)}></SomeWebCompoent>
)
```

because in react that closure will be updated every render

Reply:
reverted logic back to original

Reply:
Resolving because I think the behavior you pointed out is accounted for in 'react philosophy'.

A fresh closure would thrash the dom in react and I believe the same behavior should be expected in ours as well.